clippy_lints/src/doc/broken_link.rs

```diff
@@ -0,0 +1,172 @@
+                continue;
+            }
+
+            match &self.state {
+                None => {
+                    if c == '[' {
+                        self.state = Some(State::ProcessingLinkText);
+                        // +3 skips the opening delimiter
+                        self.active_pos_start = attr_span.lo().0 + u32::try_from(pos).unwrap() + 3;
+                        self.active_span = Some(attr_span);
+                    }
+                },
+                Some(State::ProcessingLinkText) => {
+                    if c == ']' {
+                        self.state = Some(State::ProcessedLinkText);
+                    }
+                },
+                Some(State::ProcessedLinkText) => {
+                    if c == '(' {
+                        self.state = Some(State::ProcessingLinkUrl(UrlState::Empty));
+                    } else {
+                        // not a real link, start lookup over again
+                        self.reset_lookup();
+                    }
+                },
+                Some(State::ProcessingLinkUrl(url_state)) => {
+                    if c == ')' {
+                        // record full broken link tag
+                        if let UrlState::FilledBrokenMultipleLines = url_state {
+                            // +3 skips the opening delimiter and +1 to include the closing parethesis
+                            let pos_end = attr_span.lo().0 + u32::try_from(pos).unwrap() + 4;
+                            self.record_broken_link(pos_end, BrokenLinkReason::MultipleLines);
+                            self.reset_lookup();
+                        }
+                        self.reset_lookup();
+                        continue;
+                    }
+
+                    if !c.is_whitespace() {
+                        if reading_link_url_new_line {
+                            // It was reading a link url which was entirely in a single line, but a new char
+                            // was found in this new line which turned the url into a broken state.
+                            self.state = Some(State::ProcessingLinkUrl(UrlState::FilledBrokenMultipleLines));
+                            continue;
+                        }
+
+                        self.state = Some(State::ProcessingLinkUrl(UrlState::FilledEntireSingleLine));
+                    }
+                },
+            };
+        }
+    }
+
+    fn reset_lookup(&mut self) {
+        self.state = None;
+        self.active_span = None;
+        self.active_pos_start = 0;
+    }
+
+    fn record_broken_link(&mut self, pos_end: u32, reason: BrokenLinkReason) {
+        if let Some(attr_span) = self.active_span {
+            let start = BytePos(self.active_pos_start);
+            let end = BytePos(pos_end);
+
+            let span = Span::new(start, end, attr_span.ctxt(), attr_span.parent());
+
+            self.broken_links.push(BrokenLink { reason, span });
+        }
+    }
```

Comment:
This should just be raising the lint directly rather than storing them first.

Reply:
done